fog/types/src/view.rs

```diff
@@ -236,9 +236,14 @@ pub struct TxOutRecord {
     pub tx_out_e_memo_data: Vec<u8>,
 
     /// The masked token id associated to the amount field in the TxOut that
-    /// was recovered
+    /// was recovered. This is used when the masked amount has version 1.
     #[prost(bytes, tag = "10")]
     pub tx_out_amount_masked_token_id: Vec<u8>,
+
+    /// The masked token id associated to the amount field in the TxOut that
+    /// was recovered. This is used when the masked amount has version 2.
+    #[prost(bytes, tag = "11")]
+    pub tx_out_amount_v2_masked_token_id: Vec<u8>,
```

Comment:
What's motivating the choice to make a separate field for a v2 token id?

Reply:
That totally makes sense. I think it read a little odd, but I don't have a better suggestion at the moment.